doc/source/serve/http-guide.md

```diff
@@ -41,6 +41,32 @@ Often for ML models, you just need the API to accept a `numpy` array. You can us
 Serve provides a library of HTTP adapters to help you avoid boilerplate code. The [later section](serve-http-adapters) dives deeper into how these works.
 ```
 
+(serve-handling-client-disconnects-http)=
+### Handling client disconnects
+
+When an HTTP client disconnects before receiving a response, Serve cancels its in-flight request:
+
+- If the proxy hasn't yet sent the request to a replica, the request is simply dropped.
+- If the request has been sent to a replica, Serve attempts to interrupt the replica and cancel the request. When the replica enters an `await` statement, Serve raises an `asyncio.CancelledError`. Handle this exception in a try-except block to customize your deployment's behavior when a request is cancelled:
+
+```{literalinclude} doc_code/http_guide/http_guide.py
+:start-after: __start_basic_disconnect__
+:end-before: __end_basic_disconnect__
+:language: python
+```
+
+If there are no remaining `await` statements in the deployment's code before the request completes, the replica processes the request as usual, sends the response back to the proxy, and the proxy discards the response. Use `await` statements for blocking operations in a deployment, so in-flight requests can be cancelled in the deployment without waiting for the blocking operation to complete.
+
+Serve implements cascading cancellations. If a deployment is `awaiting` a response from another Ray task, Ray actor, or Serve deployment, Serve raises the `asyncio.CancelledError` in the `awaiting` deployment as well as the downstream task, actor, or deployment. This cancellation cascades through any chain of tasks, actors, or deployments that are `awaiting` another task, actor, or deployment. To block a cancellation from cascading to a downstream task, actor, or deployment, use `asyncio.shield`:
```

Comment:
the cancellation in downstreams is independent of whether or not you are `await`ing the response. it will happen to any downstream task/actor call you've made

Reply:
Oh I didn't know that. I'll change the wording.

Reply:
Can be more concise:

> Cancellation will cascade to any downstream deployment handle, task, or actor calls that were spawned in the deployment request handling method. These can handle the `asyncio.CancelledError` in the same was as the ingress deployment.

And once again, `shield` isn't specific to cascading. Mention it above when describing how to handle the error

Reply:
Oh, I think I was looking at the outdated version. But I think what I wrote in the quote block above should suffice.

Reply:
Sounds good, I'll switch tot he block you wrote. Does the updated explanation about `shield` make sense?